### PR TITLE
feat(cache): replace ehcache by hazelcast

### PR DESCRIPTION
**Pull request by gcusnieux**
fix: gravitee-io/issues#599

### CODE REVIEW

File: src/main/java/io/gravitee/resource/cache/CacheResource.java [hunk 1]
```diff
@@ -15,58 +15,97 @@
  */
 package io.gravitee.resource.cache;
 
+import com.hazelcast.config.Config;
+import com.hazelcast.config.EvictionPolicy;
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.config.MaxSizeConfig;
+import com.hazelcast.core.Hazelcast;
```
Comment (ytvnr) [inline src/main/java/io/gravitee/resource/cache/CacheResource.java:22]: i think these imports are not used

File: src/main/java/io/gravitee/resource/cache/CacheResource.java [hunk 2]
```diff
@@ -15,58 +15,97 @@
...
+    private HazelcastInstance hazelcastInstance;
     private Cache cache;
+    private ApplicationContext applicationContext;
+
+    protected static MapConfig updateMapConfig(HazelcastInstance hazelcastInstance, CacheResourceConfiguration configuration) {
+        Config config = hazelcastInstance.getConfig();
+        MapConfig mapConfig = config.getMapConfig(configuration.getName());
+
+        if (
+            configuration.getMaxEntriesLocalHeap() != 0 && configuration.getMaxEntriesLocalHeap() < mapConfig.getMaxSizeConfig().getSize()
```
Comment (ytvnr) [inline src/main/java/io/gravitee/resource/cache/CacheResource.java:53]: weird formatting

**Review by ytvnr — APPROVED** [review]
uneused import and formatting issue

File: src/main/java/io/gravitee/resource/cache/CacheResource.java [hunk 3]
```diff
@@ -15,58 +15,97 @@
...
  * @author David BRASSELY (david at gravitee.io)
  * @author GraviteeSource Team
  */
-public class CacheResource extends AbstractConfigurableResource<CacheResourceConfiguration> {
+public class CacheResource extends AbstractConfigurableResource<CacheResourceConfiguration> implements ApplicationContextAware {
 
     private final Logger LOGGER = LoggerFactory.getLogger(CacheResource.class);
 
-    private CacheManager cacheManager;
+    private String hazelcastConfigFilePath;
```
Comment (ytvnr) [inline src/main/java/io/gravitee/resource/cache/CacheResource.java:42]: useless

**Review by phiz71 — APPROVED** [review]

**Review by phiz71 — APPROVED** [review]

**Pull request merged**